lib/launchers/amz.rb

```diff
@@ -471,6 +480,63 @@ def get_regions
       client_ec2.describe_regions.to_a[0][0]
     end
 
+    def default_zone
+      config[:install_base_domain].sub(/\.$/,"") + "."
+    end
+
+    def default_zone_id
+      unless @default_zone_id
+        zone ||= client_r53.list_hosted_zones.hosted_zones.find { |z|
+          z.name == default_zone
+        }
+        unless zone
+          raise "can't find the configured default zone in AWS"
+        end
+        @default_zone_id = zone.id.sub(%r{.*/(.*)$}, "\\1")
+      end
+      return @default_zone_id
+    end
+
+    # @param [Hash] change might be something like
+    #   [{
+    #     :action=>"UPSERT",
+    #     :resource_record_set=> {
+    #       :name=>"myhost.qe.devcluster.openshift.com",
+    #       :resource_records=> [{:value=>"192.0.2.44"}],
+    #       :ttl=>60,
+    #       :type=>"A"
+    #     }
+    #   }]
+    # @see https://docs.aws.amazon.com/sdk-for-ruby/v2/api/Aws/Route53/Client.html#change_resource_record_sets-instance_method
+    def change_resource_record_sets(zone_id: nil, changes:)
+      zone_id ||= default_zone_id
+      client_r53.change_resource_record_sets(
+        hosted_zone_id: zone_id,
+        change_batch: { changes: changes }
+      )
+    end
+
+    def list_resource_record_sets(zone_id: nil)
+      zone_id ||= default_zone_id
+      res = []
+      client_r53.list_resource_record_sets(hosted_zone_id: zone_id).each_page { |p|
+        res.concat p.resource_record_sets.map(&:to_h)
+      }
+      return res
+    end
+
+    # @param [Regexp] re
+    # @note be very careful to avoid interfering regular expressions
+    # example: delete_resource_records_re(/my-hostname/)
+    def delete_resource_records_re(re, zone_id: nil)
```

Comment:
SYNC vs ASYNC as above

Reply:
Tried today and based on the speed it is definitely an ASYNC operation right now which isn't the best in this case from my PoV.

Reply:
For deleting records, I think it doesn't matter. It is for clean-up only purposes anyway so we shouldn't care.

Reply:
btw knowing that operation completed, doesn't mean that item will be unresolvable immediately, it will depend on TTL.

Reply:
In case it is needed for something, we can also implement waiting until TTL finished to make sure nothing will resolve the name after method returned. Still I wonder whether that is needed in our use case.

Reply:
I'm not fully for the statement we shouldn't care - I'm not sure if requests are somehow serialized on AWS side, but if we spin-down and spin-up a cluster fast enough, this operations can be interleaved and so executed in a wrong ordering... - we create a new records (they overwrite the older values because we use `UPSERT` operation) but later we delete them just because the previous delete operation takes so much time. Is it just a hypothetical case? Don't know...

TTL value as of 60s is more legible for me than 100s general time period, it doesn't matter for installer more I guess.

Reply:
Wrong ordering can't happen. Records are deleted when full record is provided. The upserted record will have different IPs thus it shouldn't be deleted.

Actually DNS providers give a one hour or so default TTL IIRC. To me 60 seconds seems too aggressive leading to many more DNS queries. But on the other hand we overwrite records so 1 hour is too much. 3 minutes was my gut feeling not to be too bad to the DNS. Lets keep it 3 minutes unless we observe some issue with that value.

Reply:
Fine, 3min -> 180s (not 100), was a typo originally?

Reply:
Just a note - UPSERTed records don't have a different IPs every time (as I observed), at least some of the records... but you are right, we are forced to provide an exact record to be deleted, it is now hidden from user's perspective only (to a RegExp).